### ISSUE TITLE
New setting - do not save `saveUser`, `saveUserFull`, `author` in project files (qgs) for anonymity

### ISSUE THREAD

**Issue by MrChebur** [qgis/QGIS#55421]
### Feature description

Anyone can open and read the qgs/qgz project file and see the author's name, surname and user name.
It would be nice to add the ability to hide this information when saving project files.
In example in the `Options` => `General` => `Project Files` => `Don't save user name in the project file`

```
<qgis projectname="" saveDateTime="2023-11-27T10:09:11" version="3.34.0-Prizren" saveUserFull="Name Surname" saveUser="Username">
...
<author>Name Surname</author>
```

**Comment by agiudiceandrea** [comment]
@MrChebur, you can change such behaviour setting to `true` the `core/projects/anonymize_new_projects` and `core/projects/anonymize_saved_projects` options (https://docs.qgis.org/3.28/en/docs/user_manual/introduction/qgis_configuration.html#advanced-settings).

For some background: https://github.com/qgis/QGIS/pull/33843, https://github.com/qgis/QGIS/pull/33870, https://github.com/qgis/QGIS-Enhancement-Proposals/issues/115, https://github.com/qgis/QGIS/pull/6635, https://lists.osgeo.org/pipermail/qgis-developer/2021-March/063355.html, https://lists.osgeo.org/pipermail/qgis-it-user/2021-March/009056.html.

**Comment by MrChebur** [comment]
Thanks, @agiudiceandrea , the `saveUser`, `saveUserFull` tags are now missing in the project file, but the <author></author> tag is still present in the project with the settings you specified. At least in Windows 10, it contains the username, which is often a last name and first name.

**Comment by mrpiggi** [comment]
It would be great, if this would also be used or could be configured when using `pyqgis` but for 3.28, those settings are completely ignored. Saving a project with the desktop app anonymously is fine but with

```python
from qgis.core import QgsApplication
from qgis.core import QgsProject

qgs = QgsApplication([], False)
qgs.initQgis()
project = QgsProject.instance()
project.read('some-project.qgs')
project.write()
qgs.exitQgis()
```
`saveUser`, `saveUserFull` and `saveDateTime` re-emerge

**Comment by agiudiceandrea** [comment]
@mrpiggi, it looks like you have forgot to properly set the `projects/anonymize_new_projects` and `projects/anonymize_saved_projects` options in the `core` scope of the QgsProject instance created.

**Comment by mrpiggi** [comment]
Yes, I wondered if there are such settings. But as those are made via `Settings | Options | Advanced` I assumed, those could be found in `QgsApplication`. But it turns out, that these settings are rather stored within the user profile.

Currently, it does not really bother me as the actual reason for using this feature was to minimize changes in the `*.qgs` file in order to use it with git. But as the XML attributes are shuffled anyway every time saving, this is for now not really an issue. At least for me.
